### DISCUSSION TITLE
How to give platform neutral name for runfiles specified throug data attribute

### DISCUSSION THREAD

**Discussion by v8App** [bazelbuild/bazel#19816] [General]
I have a library that needs to include some files via data attribute that are divided into platform specific directories. Only the platform that is being built is include when run and i'd like for it to have a platform neutral name to refernce the real location when using Rlocation.

example in if i include lib/win-64-release/icu.dat it has the platform in the name used to look it up. I'd like to be able to just look it up as lib/icu.dat so the test code doesn't have to worry about what platform path to use.

I have a customzied version of the c++ runfiles that added a wildcard operators so it could just wildcard out the platform specific part but i'd like to drop the customization if there is a way to change the name used to look up the real location from the manifest file.

**Comment by fmeum** [comment]
You could pass the runfiles path for the dependency into the test via `env = {"ICU_DAT_RLOCATIONPATH": "$(rlocationpath //my/platform/specific:dep)"}` and then into the standard C++ `Rlocation` function to look up the full path.

**Reply by v8App** [reply]
Can't see to get this to quite work. When i i do 'V8_ICU_DATA':"$(rlocationpath //third_party/v8:v8)", it gives expression expands to no files even though it's in the data attribute. Though i'd like to have separate ones for the the two data files.

**Comment by v8App** [comment]
I finally got to work but here some additional notes to get it to work. I still hade to include the file in the cc_test's data attribute even though it was specified in the v8 lib's data attribute. I also aliased it so the alias with selects would resolve the correct platform specific path since you still needed.

Without the file listed in the tests data attribute i would get the error expression is not a declared prerequisite of this rule.
Not sure why this was as without the env vars the runfiles manifest had the v8 file in there.

**Reply by fmeum** [reply]
> Without the file listed in the tests data attribute i would get the error expression is not a declared prerequisite of this rule.
> Not sure why this was as without the env vars the runfiles manifest had the v8 file in there.

You got the error because you can only use location expansions such as `$(rlocationpath ...)` on *direct* dependencies of your target. The file should be there regardless, but if you want to get its path, you need to add it to `data` of the target referring to it in this way.
